### DISCUSSION TITLE
Tizen DPAD

### DISCUSSION THREAD

**Discussion by michalss** [AvaloniaUI/Avalonia#13801] [General]
Hello,

With upcoming Tizen, would be also DPAD supported for android ? Or would be supported at all ?

Regards

**Comment by maxkatz6** [comment]
No, gamepad/controlls support is not implemented unless it's mapped to the keyboard keys by the platform. So you most likely would need to handle these events manually. See https://github.com/AvaloniaUI/Avalonia/issues/6945

cc @OmidID

**Reply by michalss** [reply]
OK, but im sorry to say it, What is the point to support purely TV os without his primary control focus as Directional PAD ? It does not make any sense to me at all.

**Reply by OmidID** [reply]
Hello. 
Yes. Already DPad supported by Android and also by Tizen. In fact the Dpad act as a keyboard and it is supported by by almost all platforms. You can see in the demo i use DPad ( https://x.com/AvaloniaUI/status/1671617269774929920 )

@maxkatz6  Dpad is not gamepad/controller. basically DPad is the arrow key on remote control. https://developer.android.com/design/ui/tv/guides/foundations/navigation-on-tv#controller

**Reply by michalss** [reply]
Ohh love it. Can you please share some sample code for android? Im waiting for this a long time... I was not able to find any sample code for Android DPAD... :( Or is this will come together with Tizen ?

**Reply by maxkatz6** [reply]
> @maxkatz6 Dpad is not gamepad/controller. basically DPad is the arrow key on remote control.

The Remote control is a controller, and dpad is part of it, if we are speaking API-terms wise of other frameworks. Gamepad is just another type of controllers.
But yeah, good to know it works with keydown events already.

**Reply by OmidID** [reply]
Yes. Because the tv remote control act as keyboard it is working anyway.

@michalss you do not need any demo or sample app. just try to get events of keyboard.

**Comment by michalss** [comment]
> Yes. Because the tv remote control act as keyboard it is working anyway.
> 
> @michalss you do not need any demo or sample app. just try to get events of keyboard.

hmm not sure how exactly. still can you please be so kind and make simple POC for us ?

**Reply by OmidID** [reply]
You do not need any sample app or POC. As I explained already, just subscribe to key events. It is very simple. 
The control that already supports the arrow key same as TabControl they switch with left and right arrow keys.

![image](https://github.com/AvaloniaUI/Avalonia/assets/956077/6edac82d-6abe-47f4-be6e-eee502f80df4)


Run the ControlCatalog on the Android TV and try it.

https://github.com/AvaloniaUI/Avalonia/assets/956077/2e10b333-e8b5-4baf-8cf4-2d3d0dc1795e

**Reply by michalss** [reply]
still can you please provide me with some example ? Im not sure how to handle this... pls

**Reply by OmidID** [reply]
Hello @michalss ,

DPAD should function well in Avalonia v11.1.0 and higher.

https://github.com/AvaloniaUI/Avalonia/pull/13947
